src/hotspot/share/interpreter/rewriter.cpp

```diff
@@ -185,11 +185,11 @@ void Rewriter::rewrite_field_reference(address bcp, int offset, bool reverse) {
   if (!reverse) {
     int cp_index = Bytes::get_Java_u2(p);
     int field_entry_index = _cp_map.at(cp_index);
-    Bytes::put_native_u2(p, field_entry_index);
+    Bytes::put_native_u2(p, (u2)field_entry_index);
   } else {
     int field_entry_index = Bytes::get_native_u2(p);
     int pool_index = _initialized_field_entries.at(field_entry_index).constant_pool_index();
-    Bytes::put_Java_u2(p, pool_index);
+    Bytes::put_Java_u2(p, (u2)pool_index);
```

Comment:
I guess this isn't using checked_cast because similar code elsewhere in this file uses this style, but it means we can silently do the wrong thing.  Is there a separate RFE to remove as many of these unchecked casts as possible?

Reply:
We could rewrite _initialized_field_entries to be an array of u2 and not need the cast, but that changed felt out of scope for this fix.